feat(rocsparse): add rocsparse_spmat_scale sparse matrix scalin (#9729)
ROCm/rocm-libraries: projects/rocsparse/CHANGELOG.md

## (Unreleased) rocSPARSE 5.0.0

### Added

> * Added the `rocsparse_spmat_scale` generic routine for sparse matrix scaling (`C = alpha * A`), where `C` shares the sparsity pattern of `A` and its values are scaled by `alpha`. The COO, COO AoS, CSR, CSC, BSR, ELL, Blocked ELL, and SELL formats are supported, in-place operation is allowed, and `alpha` is passed as a self-describing scalar dense vector descriptor that may live in host or device memory. No temporary storage buffer is required.

Reviewer: ```suggestion
* Added the `rocsparse_spmat_scale` generic routine for sparse matrix scaling (`C = alpha * A`), where `C` shares the sparsity pattern of `A`, whose values are scaled by `alpha`. The COO, COO AoS, CSR, CSC, BSR, ELL, Blocked ELL, and SELL formats are supported. In-place operation is allowed, and `alpha` is passed as a self-describing scalar dense vector descriptor that can live in host or device memory. No temporary storage buffer is required.
```

Author: Done, thanks!

> * Added the `rocsparse_spmat_scale` generic routine for sparse matrix scaling (`C = alpha * A`). It writes `C`'s values as `alpha` times `A`'s values and does not copy the sparsity pattern (`C` is assumed to already describe the same pattern as `A`); in-place operation (`C == A`) is supported. The COO, COO AoS, CSR, CSC, BSR, ELL, Blocked ELL, and SELL formats are supported, and `alpha` is passed as a self-describing scalar dense vector descriptor that can live in host or device memory. No temporary storage buffer is required.

Reviewer: ```suggestion
* Added the `rocsparse_spmat_scale` generic routine for sparse matrix scaling (`C = alpha * A`). It writes to `C` `alpha` times the values of `A` and does not copy the sparsity pattern (`C` is assumed to already have the same sparsity pattern as `A`). `alpha` is passed as a self-describing scalar dense vector descriptor that can reside in host or device memory, so no temporary storage buffer is required.  In-place operation (`C == A`) is supported.  COO, COO AoS, CSR, CSC, BSR, ELL, Blocked ELL, and SELL formats are supported.
```

Author: Applied, thanks!
